src/game/default/g_main.c

```diff
@@ -938,6 +948,119 @@ void G_Shutdown(void) {
 	gi.FreeTag(MEM_TAG_GAME);
 }
 
+void G_CallTimeOut(g_entity_t *ent) {
+
+	g_level.match_status |= MSTAT_TIMEOUT;
+	g_level.timeout_caller = ent;
+	g_level.timeout_time = g_level.time + (g_timeout_time->integer * 1000);
+	g_level.timeout_frame = g_level.frame_num;
+	
+	// lock players
+	for (int32_t i = 1; i < sv_max_clients->integer; i++) {
+		if (g_game.entities[i].client->locals.persistent.team) {
+			g_game.entities[i].client->ps.pm_state.type = PM_FREEZE;
+		}
+	}
+	
+	gi.BroadcastPrint(PRINT_HIGH, "%s called a timeout, play with resume in %s\n", 
+		ent->client->locals.persistent.net_name, G_FormatTime(g_timeout_time->integer * 1000));
+}
+
+void G_CallTimeIn(void) {
+	
+	g_level.frame_num = g_level.timeout_frame; // where we were before timeout
+	
+	// unlock players
+	for (int32_t i = 1; i < sv_max_clients->integer; i++) {
+		if (g_game.entities[i].client->locals.persistent.team) {
+			g_game.entities[i].client->ps.pm_state.type = PM_NORMAL;
+		}
```

Comment:
I think it's just another line or two in G_ClientConnect / G_ClientBegin / G_ClientRespawn ;)